### ISSUE TITLE
Get all symbols

### ISSUE THREAD

**Issue by TJStienstra** [TJStienstra/brim#112]
It would be nice if there was a method to quickly get all declared symbols, e.g. `BrimBase.get_all_symbols()`. This is a relatively easy feature to implement as it only requires a simple traversal, where each object returns a set of its own symbols.

**Comment by uinone** [comment]
Hello, I am a student trying to contribute for the first time. I'm asking a question because I don't quite understand the issue. Looking at the issue content, it looks like I need to simply create a static method that returns self.symbol. Is my understanding correct?

**Comment by TJStienstra** [comment]
Each component in BRiM has an attribute `symbols`, which is a dictionary mapping descriptive names to the symbol objects, e.g. `{"radius": self.Symbol(f"{self.name}_radius")}`. This new method should gather all the defined symbols in a model and return it as a set. The bigger picture is that a user can quickly check if (s)he forgot to define any symbols. I would expect the function to have the following format, but if you have better ideas feel free to make suggestions:
```py
def get_all_symbols(self) -> set[sympy.Basic]:
     """Get all defined symbols."""
     # A tree traversal of the submodels, connections, and load groups to gather all symbols.
     # I would personally use a similar traversal as BrimBase.get_description
     # Return the set of symbols.
```

**Comment by uinone** [comment]
Thanks for the explanation! I'll try my best.

**Comment by TJStienstra** [comment]
Due to inactivity, this issue will be solved in #111
